### PR TITLE
chore: upgrade to Node.js v20

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Upgrade to Node.js v20.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Upgraded to Node.js v20.

### CODE REVIEW

**Comment by sparecycles** [comment]
Can we expect that this will support the [`import('node:module').register`](https://nodejs.org/dist/latest-v20.x/docs/api/module.html#moduleregisterspecifier-parenturl-options) API in the host? 🤞

**Comment by codebytere** [comment]
Filed https://github.com/electron/electron/issues/40626

File: patches/node/chore_remove_allcan_read_write_from_setaccessor.patch [hunk 1]
```diff
@@ -0,0 +1,60 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Shelley Vohr <shelley.vohr@gmail.com>
+Date: Wed, 15 Nov 2023 12:09:38 +0100
+Subject: chore: remove AllCan Read/Write from SetAccessor
+
+Refs https://chromium-review.googlesource.com/c/v8/v8/+/5006387
+
+This patch can be removed when Node.js upgrades to a version of V8
```
Comment (zcbenz) [inline patches/node/chore_remove_allcan_read_write_from_setaccessor.patch:8]: You can backport this change to reduce future conflicts: https://github.com/nodejs/node/pull/50709

File: shell/common/node_util.cc [hunk 2]
```diff
@@ -17,9 +17,11 @@ v8::MaybeLocal<v8::Value> CompileAndCall(
     node::Environment* optional_env) {
```
Comment (zcbenz) [inline shell/common/node_util.cc:17]: The `optional_env` seems to be no longer used.

File: shell/common/node_util.cc [hunk 3]
```diff
@@ -17,9 +17,11 @@ v8::MaybeLocal<v8::Value> CompileAndCall(
...
   v8::Isolate* isolate = context->GetIsolate();
   v8::TryCatch try_catch(isolate);
+  thread_local node::builtins::BuiltinLoader builtin_loader;
```
Comment (zcbenz) [inline shell/common/node_util.cc:20]: There is a `builtin_loader()` getter in `node::Environment`.

**Review by zcbenz — APPROVED** [review]

**Review by jkleinsc — COMMENTED** [review]
API LGTM

**Review by ckerr — COMMENTED** [review]
API LGTM

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**